source/index-management/tests/README.rst

```diff
@@ -45,4 +44,154 @@ Tests
+
+Setup
+~~~~~
+
+These tests must run against an Atlas cluster with a 7.0+ server.  Tools are available drivers-evergreen-tools which can setup and teardown
+Atlas clusters.  To ensure that the Atlas cluster is cleaned up after each CI run, drivers should configure evergreen to run these tests 
+as a part of a task group.  Be sure that the cluster gets torn down! 
+
+When working locally on these tests, the same Atlas setup and teardown scripts can be used locally to provision a cluster for development.
+
+Case 1: Driver can successfully create and list search indexes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create a new search index on ``coll0`` with the ``createSearchIndex`` helper.  Use the following definition:
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command returns the name of the index: ``"test-search-index"``.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied and store the value in a variable ``index``:
+   1. An index with the ``name`` of ``test-search-index`` is present and the index has a field ``queryable`` with a value of ``true``.
+
+#. Assert that ``index`` has a property ``mappings`` whose value is ``{ dynamic: false }``
+
+Case 2: Driver can successfully create multiple indexes in batch
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Create a collection with a randomly generated name (referred to as ``coll0``).
+#. Create two new search indexes on ``coll0`` with the ``createSearchIndexes`` helper.  Use the following
+   definitions when creating the indexes.  These definitions are referred to as ``indexDefinitions``.
+
+  .. code:: typescript
+
+    {
+      name: 'test-search-index-1',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+    {
+      name: 'test-search-index-2',
+      definition: {
+        mappings: { dynamic: false }
+      }
+    }
+
+#. Assert that the command returns an array containing the new indexes' names: ``["test-search-index-1", "test-search-index-2"]``.
+#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied.  Store
+   the result in ``indexes``.
+   1. An index with the ``name`` of ``test-search-index-1`` is present and index has a field ``queryable`` with the value of ``true``.
+   2. An index with the ``name`` of ``test-search-index-2`` is present and index has a field ``queryable`` with the value of ``true``.
+#. For each ``index`` in ``indexDefinitions``
+   1. Find the matching index definition in ``indexes`` by matching on ``index.name``.  If no index exists, raise an error.
+   2. Assert that the matching index ``mappings``, whose value is ``{ dynamic: false }``
```

Comment:
```suggestion
#. Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied.
   1. An index with the ``name`` of ``test-search-index-1`` is present and index has a field ``queryable`` with the value of ``true``. Store result in ``index1``.
   2. An index with the ``name`` of ``test-search-index-2`` is present and index has a field ``queryable`` with the value of ``true``. Store result in ``index2``.
#. Assert that ``index1`` and ``index2`` have the property ``latestDefinition`` whose value is ``{ "mappings" : { "dynamic" : false } }``
```

Suggest using separate `index1` and `index2` variables to simplify test.

Suggest updating assert to use `latestDefinition` to match returned document. Here is an example of the documents returned in the C driver testing:

```
  { "id" : "64c15fc9a199d3199e27ad70", "name" : "test-search-index-1", "status" : "READY", "queryable" : true, "latestDefinition" : { "mappings" : { "dynamic" : false } } }
  { "id" : "64c15fc9a199d3199e27ad71", "name" : "test-search-index-2", "status" : "READY", "queryable" : true, "latestDefinition" : { "mappings" : { "dynamic" : false } } }
```